web/packages/teleterm/src/services/tshd/vnet.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import grpc from '@grpc/grpc-js';
+import * as vnetServiceProtobuf from 'gen-proto-ts/teleport/lib/teleterm/vnet/v1/vnet_service_pb.client';
+import { GrpcTransport } from '@protobuf-ts/grpc-transport';
+
+import Logger from 'teleterm/logger';
+
+import { loggingInterceptor } from './interceptors';
+import { CloneableClient, cloneClient } from './cloneableClient';
+
+export type VnetServiceClient =
+  CloneableClient<vnetServiceProtobuf.VnetServiceClient>;
+
+export function createVnetClient(
+  addr: string,
+  credentials: grpc.ChannelCredentials
+): VnetServiceClient {
+  const logger = new Logger('vnet');
+  const transport = new GrpcTransport({
```

Comment:
Oh yeah, good point, we should probably reuse them.
I can update the interceptor to use `tshd` as the name but I'll make sure it logs the service name as well.

https://learn.microsoft.com/en-us/aspnet/core/grpc/performance?view=aspnetcore-7.0#reuse-grpc-channels

Reply:
>I can update the interceptor to use tshd as the name but I'll make sure it logs the service name as well.

Cool!

Reply:
>Now it's just:

```
[tshd] info: receive teleport.lib.teleterm.v1.TerminalService
GetCluster { … }
```
I don't have a strong opinion, but this looks fine.
One more thing that we could do to improve readability would be to not stringify objects in the dev tools:
<img width="661" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/15c7cdc8-d8b8-4717-b218-89696014d3c8">
I was experimenting with this recently.


 >Also, I noticed that the pin sensitive property filters out pinnedResources too. 💀 idk, I guess the easiest way to solve this would be to make a list with exceptions.

Or simply match full property names, I'm not sure if we need to use `.includes()` 🤔

Reply:
I thought about not stringifying objects some time ago, but doesn't this conflicts with logging to a file or something? I remember I tried it a long time ago but I ran into some problems.

As for full property names, yeah, that's an option as well. I guess I initially wanted us to do substring matching just to be safe.

Reply:
>I thought about not stringifying objects some time ago, but doesn't this conflicts with logging to a file or something? 

No, I use different "logging transports" for file and for dev tools. I will open a PR for that in some spare time
(and fix the bug with filtering "pinnedResources").